gn3/db/rdf.py

```diff
@@ -41,40 +41,57 @@ def query_and_frame(query: str, context: dict, endpoint: str) -> dict:
 
 def get_wiki_entries_by_symbol(symbol: str, sparql_uri: str) -> dict:
     """Fetch all the Wiki entries using the symbol"""
-    # FIXME: Get the latest VersionId of a comment.
+    # This query uses a sub-query to fetch the latest comment by the
+    # version id.
     query = Template("""
 $prefix
 
 CONSTRUCT {
-    _:node rdfs:label '$symbol';
-           gnt:reason ?reason ;
-           gnt:species ?species ;
-           dct:references ?pmid ;
-           foaf:homepage ?weburl ;
-           rdfs:comment ?wikientry ;
-           foaf:mbox ?email ;
-           gnt:initial ?usercode ;
-           gnt:belongsToCategory ?category ;
-           gnt:hasVersion ?versionId
+    ?uid rdfs:label ?symbolName;
+         gnt:reason ?reason ;
+         gnt:species ?species ;
+         dct:references ?pmid ;
+         foaf:homepage ?weburl ;
+         rdfs:comment ?comment ;
+         foaf:mbox ?email ;
+         gnt:initial ?usercode ;
+         gnt:belongsToCategory ?category ;
+         gnt:hasVersion ?versionId ;
+         dct:created ?created ;
+         dct:identifier ?identifier .
 } WHERE {
-    ?symbolId rdfs:comment _:node ;
-              rdfs:label '$symbol' .
-    _:node rdf:type gnc:GNWikiEntry ;
-           dct:hasVersion "0"^^xsd:int ;
-           dct:hasVersion ?version ;
-           rdfs:comment ?wikientry .
-    OPTIONAL { _:node gnt:reason ?reason } .
+    ?symbolId rdfs:label ?symbolName .
+    ?uid rdfs:comment ?comment ;
+         gnt:symbol ?symbolId ;
+         rdf:type gnc:GNWikiEntry ;
+         dct:created ?createTime .
+    FILTER ( LCASE(?symbolName) = LCASE('$symbol') ) .
+    {
+        SELECT (MAX(?vers) AS ?max) ?id_ WHERE {
```

Comment:
I'm not too sure about this approach performance wise. If this behaves similar to sql, we won't be able to optimize this since the subquery will have to run n times. Can we try this approach instead:
```
SELECT distinct_cols, MAX(?vers), other_cols FROM tables GROUP BY distinct_cols
```
I've seen a query that may do something similar here: https://en.wikibooks.org/wiki/SPARQL/Aggregate_functions

Reply:
Something to consider in another PR.